internal/policies/mount/internal_test.go

```diff
@@ -0,0 +1,88 @@
+package mount
+
+import (
+	"flag"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+var Update bool
+
+func TestWriteMountsFile(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		path    string
+		entries string
+
+		readOnlyDir bool
+
+		wantErr bool
+	}{
+		// Single entry cases.
+		"write mounts file with one entry with one value":        {entries: "one entry with one value"},
+		"write mounts file with one entry with multiple values":  {entries: "one entry with multiple values"},
+		"write mounts file with one entry with repeatead values": {entries: "one entry with repeatead values"},
+
+		// Multiple entries cases.
+		"write mounts file with multiple entries with one value":       {entries: "multiple entries with one value"},
+		"write mounts file with multiple entries with multiple values": {entries: "multiple entries with multiple values"},
+		"write mounts file with multiple entries with the same value":  {entries: "multiple entries with the same value"},
+		"write mounts file with multiple entries with repeated values": {entries: "multiple entries with repeated values"},
+
+		// Badly formatted entries.
+		"write mounts file trimming whitespaces":           {entries: "entry with linebreaks and spaces"},
```

Comment:
I wouldn’t mix linebreaks and space (to minimize the amount of concern in this test).